Fedora/29

diehlpk opened the pull request.

diehlpk committed:
Hpx builds with gcc, mpich, and openmpi, see https://koji.fedoraproject.org/koji/taskinfo?taskID=30764663

Following packages are generated:

hpx-1.2.0-0.1.rc1.i686.rpm
hpx-debuginfo-1.2.0-0.1.rc1.i686.rpm
hpx-debugsource-1.2.0-0.1.rc1.i686.rpm
hpx-devel-1.2.0-0.1.rc1.i686.rpm
hpx-examples-1.2.0-0.1.rc1.i686.rpm
hpx-examples-debuginfo-1.2.0-0.1.rc1.i686.rpm
hpx-mpich-1.2.0-0.1.rc1.i686.rpm
hpx-mpich-debuginfo-1.2.0-0.1.rc1.i686.rpm
hpx-mpich-devel-1.2.0-0.1.rc1.i686.rpm
hpx-mpich-examples-1.2.0-0.1.rc1.i686.rpm
hpx-mpich-examples-debuginfo-1.2.0-0.1.rc1.i686.rpm
hpx-openmpi-1.2.0-0.1.rc1.i686.rpm
hpx-openmpi-debuginfo-1.2.0-0.1.rc1.i686.rpm
hpx-openmpi-devel-1.2.0-0.1.rc1.i686.rpm
hpx-openmpi-examples-1.2.0-0.1.rc1.i686.rpm
hpx-openmpi-examples-debuginfo-1.2.0-0.1.rc1.i686.rpm

diehlpk committed:
Move lib.a to the devel package

diehlpk committed:
Merge branch 'master' into fedora/29

diehlpk committed:
Update change log

diehlpk committed:
Replace %{_libdir}/${mpi}/lib with ${MPI_LIB}

diehlpk committed:
Move lib*.so to the devel packages

diehlpk committed:
Merge branch 'fedora/29' of https://github.com/diehlpk/hpx_packaging into fedora/29

diehlpk committed:
Remove mpi packages from post and postun

diehlpk committed:
Use %ldconfig_scriptlets libs instead of post and postun

diehlpk committed:
Use ${MPI_LIB:-%{_libdir}} instead of the if statement

diehlpk committed:
Remove ${s:-.}/

junghans committed:
Fedora/hpx.spec: another slight clean up - untested (#8)

diehlpk committed:
Add license file

junghans commented:
Besides whatever `fedora-review` finds, I think it looks good now!

One last thing, we could put the duplicated description text box in a `%global` macro:
```
%global description HPX ist great bla bla
```

diehlpk commented:
For some packages we have one additional sentence, like This package contains development headers and libraries for hpx.

How could I add it to the %global description?

junghans commented:
I would just use it for the duplicated block and leave the individual last sentences where they are. See this example here: [lammps.spec](https://github.com/diehlpk/hpx_packaging/files/2568541/lammps.spec.txt)

diehlpk commented:
I will do this later today and will run fedora-review after.

diehlpk committed:
Change the parameter to On

diehlpk committed:
Add global description to reduce redundancy

diehlpk committed:
Remove warning from rpmlint (refs #7)

junghans commented on Fedora/hpx.spec:
```diff
@@ -259,24 +175,13 @@ for mpi in '' openmpi mpich ; do
-
-%postun openmpi -p /sbin/ldconfig
+%ldconfig_scriptlets libs
 
```
I think the `libs` is actually not needed!

diehlpk replied:
> On Fedora 28 and newer, no scriptlets are required when shared libraries are installed. However, the following scriptlets MAY be used, as they will simply evaluate to nothing on newer Fedora releases. 

https://fedoraproject.org/wiki/Packaging:Scriptlets#Shared_Libraries

So, do we need it at all since the package will start to be available in Fedora 29?

From my understanding, the command ist just ignored if the koji build target is f29 or I am wrong?

[Build](/https://koji.fedoraproject.org/koji/taskinfo?taskID=30780666) removing the scriptlet.

diehlpk replied:
Once this build is finished, I will run the fedora-review on it.

junghans replied:
Sorry I wasn’t clear `libs` is the package name, but we don’t have a libs package here. Keep the snippet as we might want to deploy this on F28 or EPEL7.

diehlpk replied:
Sure, I will add it again and do the fedora-review with the latest koji build this evening.

diehlpk committed:
Remove scriptlets since they are not used in fedora >= 28 anymore

diehlpk committed:
Add scriplets due to backwards portability again

diehlpk commented:
```
[Patrick@thinpad review]$ fedora-review --rpm-spec --prebuilt --name hpx 
INFO: Processing local files: hpx
INFO: Getting .spec and .srpm Urls from : Local files in /home/Patrick/review
INFO:   --> SRPM url: file:///home/Patrick/review/hpx-1.2.0-0.1.rc1.src.rpm
INFO: Using review directory: /home/Patrick/review/hpx
WARNING: No disttag found in prebuilt packages
```
Added `Release:        0.1.rc1%{?dist}`

diehlpk committed:
Add dist variable to the release

diehlpk committed:
Add instructions to run fedora-review

diehlpk committed:
Add less cpu for the ppc build to avoid compilation errors (refs #3)

junghans committed:
Fedora/f29: updates for 1.2.0 (#9)

junghans commented:
https://bugzilla.redhat.com/show_bug.cgi?id=1649874

diehlpk committed:
Reduce the cores for the i686 build and remove the rm for the
/usr/share/doc folder since this one it not generated in HPX 1.2
when the documentation is turned off.

junghans commented:
https://koji.fedoraproject.org/koji/taskinfo?taskID=30886771

junghans commented:
Hmm, i686 has a real problem:
```
In file included from /builddir/build/BUILD/hpx_1.2.0/hpx/lcos/local/packaged_continuation.hpp:22,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/lcos/future.hpp:1798,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors/default_executor.hpp:12,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors.hpp:11,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/include/parallel_executors.hpp:9,
                 from /builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:6:
/builddir/build/BUILD/hpx_1.2.0/hpx/util/deferred_call.hpp:160:5: error: 'hpx::util::detail::deferred<typename std::decay<_Tp>::type, typename std::decay<_Args>::type ...> hpx::util::deferred_call(F&&, Ts&& ...) [with F = const test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; Ts = {hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >}; typename std::decay<_Tp>::type = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>]', declared using local type 'const test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>', is used but never defined [-fpermissive]
     deferred_call(F&& f, Ts&&... vs)
     ^~~~~~~~~~~~~
In file included from /builddir/build/BUILD/hpx_1.2.0/hpx/lcos/local/packaged_continuation.hpp:22,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/lcos/future.hpp:1798,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors/default_executor.hpp:12,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/parallel/executors.hpp:11,
                 from /builddir/build/BUILD/hpx_1.2.0/hpx/include/parallel_executors.hpp:9,
                 from /builddir/build/BUILD/hpx_1.2.0/examples/resource_partitioner/async_customization.cpp:6:
/builddir/build/BUILD/hpx_1.2.0/hpx/util/deferred_call.hpp:160:5: error: 'hpx::util::detail::deferred<typename std::decay<_Tp>::type, typename std::decay<_Args>::type ...> hpx::util::deferred_call(F&&, Ts&& ...) [with F = const test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>; Ts = {hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >}; typename std::decay<_Tp>::type = test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>]', declared using local type 'const test(Executor&) [with Executor = hpx::threads::executors::guided_pool_executor_shim<hpx::threads::executors::pool_numa_hint<dummy_tag> >]::<lambda(hpx::lcos::future<hpx::util::tuple<hpx::lcos::future<long long unsigned int>, hpx::lcos::shared_future<float> > >&&)>', is used but never defined [-fpermissive]
     deferred_call(F&& f, Ts&&... vs)
     ^~~~~~~~~~~~~
```

diehlpk commented:
Yes, we are looking at it right now. At least x86 builds again :)

https://github.com/STEllAR-GROUP/hpx/issues/3549

junghans commented:
Let me know when you have a patch!

junghans commented:
OpenSuse has the same problem on i586: https://build.opensuse.org/package/show/home:cjunghans:branches:science:HPC/hpx

junghans commented:
Ok I added the patch from https://github.com/STEllAR-GROUP/hpx/pull/3551 in https://koji.fedoraproject.org/koji/taskinfo?taskID=30901371

junghans commented:
And https://build.opensuse.org/package/show/home:cjunghans:branches:science:HPC/hpx

junghans committed:
fedora/hpx.spec: include patch to build on i686 (#10)

diehlpk commented:
@junghans Is it possible to add a License header and our both names to the spec file, as for the open suse one?

junghans commented:
It sure is possible, but by the end of the day `hpx.spec` will live in the git repo for the distribution, so I would worry about that too much.

diehlpk commented:
Ok, I was thinking to keep a version in this repo here too. So we could update future changes here and only push the final working version to the fedora repo. We may update the spec file for future versions of hpx.

junghans commented:
I tried that in the past with other projects, but in Fedora the spec gets bumped from time to time by release maintenaners anyhow, so you will have to merge things manually anyhow.

junghans committed:
fedora/hpx.spec: don't build aarch64 in parallel (#12)

junghans commented:
https://pagure.io/releng/fedora-scm-requests/issue/8897

diehlpk commented:
Nice, can you add me to this repo?

junghans commented:
https://pagure.io/releng/fedora-scm-requests/issue/8902
https://pagure.io/releng/fedora-scm-requests/issue/8903
https://pagure.io/releng/fedora-scm-requests/issue/8904

junghans commented:
Once the repo is there, you will need an https://src.fedoraproject.org account!

diehlpk commented:
Already had it (https://src.fedoraproject.org/user/diehlpk)

junghans commented:
https://koji.fedoraproject.org/koji/taskinfo?taskID=30997586
https://koji.fedoraproject.org/koji/taskinfo?taskID=30997599
https://koji.fedoraproject.org/koji/taskinfo?taskID=30997601
https://koji.fedoraproject.org/koji/taskinfo?taskID=30997603

junghans commented:
> Already had it (https://src.fedoraproject.org/user/diehlpk)

There is an error: "This user must be in one of the following groups to be allowed to be added to this project: packager"

diehlpk commented:
https://koji.fedoraproject.org/koji/getfile?taskID=30997630&volume=DEFAULT&name=build.log&offset=-4000

The epl7 build failed somehow

junghans commented:
> https://koji.fedoraproject.org/koji/getfile?taskID=30997630&volume=DEFAULT&name=build.log&offset=-4000
> 
> The epl7 build failed somehow

Yup, the gcc is too old, see `root.log`.

diehlpk commented:
> > https://koji.fedoraproject.org/koji/getfile?taskID=30997630&volume=DEFAULT&name=build.log&offset=-4000
> > The epl7 build failed somehow
> 
> Yup, the gcc is too old, see `root.log`.

Ok, so we just do not provide the epl7?

diehlpk commented:
> > Already had it (https://src.fedoraproject.org/user/diehlpk)
> 
> There is an error: "This user must be in one of the following groups to be allowed to be added to this project: packager"

I reported the issue at the ticket in bugzilla.

junghans commented:
> > > https://koji.fedoraproject.org/koji/getfile?taskID=30997630&volume=DEFAULT&name=build.log&offset=-4000
> > > The epl7 build failed somehow
> > 
> > 
> > Yup, the gcc is too old, see `root.log`.
> 
> Ok, so we just do not provide the epl7?

https://lists.fedoraproject.org/archives/list/epel-devel@lists.fedoraproject.org/thread/YQLNZYT5ATELGMHNGET7QCOG3S3UIYT5/ seems related.

junghans commented:
https://bodhi.fedoraproject.org/updates/FEDORA-2018-8367209bd8
https://bodhi.fedoraproject.org/updates/FEDORA-2018-a9b53adbdc

diehlpk commented:
@junghans Can you approve tis PR so I can merge it?
